analyses/org.eclipse.tracecompass.incubator.executioncomparision.ui/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/ui/DifferentialFlameGraphView.java

```diff
@@ -0,0 +1,1261 @@
+            return fdataProviderGroup;
+        }
+
+        private void setFdataProviderGroup(@Nullable ITimeGraphDataProvider<TimeGraphEntryModel> dataProviderGroup) {
+            fdataProviderGroup = dataProviderGroup;
+        }
+
+        /**
+         * Adds a list of entries to a trace's entry list
+         *
+         * @param trace
+         *            the trace
+         * @param list
+         *            the list of time graph entries to add
+         */
+        private void addToEntryList(ITmfTrace trace, List<TimeGraphEntry> list) {
+            synchronized (fEntryListMap) {
+                List<TimeGraphEntry> entryList = fEntryListMap.get(trace);
+                if (entryList == null) {
+                    fEntryListMap.put(trace, new CopyOnWriteArrayList<>(list));
+                } else {
+                    for (TimeGraphEntry entry : list) {
+                        if (!entryList.contains(entry)) {
+                            entryList.add(entry);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private static BiFunction<ITimeEvent, Long, Map<String, String>> getTooltipResolver(ITimeGraphDataProvider<? extends TimeGraphEntryModel> provider) {
+        return (event, time) -> getTooltip(event, time, provider, false);
+
+    }
+
+    private static Map<String, String> getTooltip(ITimeEvent event, Long time, ITimeGraphDataProvider<? extends TimeGraphEntryModel> provider, boolean getActions) {
+        ITimeGraphEntry entry = event.getEntry();
+
+        if (!(entry instanceof TimeGraphEntry)) {
+            return Collections.emptyMap();
+        }
+        long entryId = ((TimeGraphEntry) entry).getEntryModel().getId();
+        IOutputElement element = null;
+        if (event instanceof TimeEvent timeEvent) {
+            element = timeEvent.getModel();
+        }
+        Map<String, Object> parameters = getFetchTooltipParameters(time, entryId, element);
+        if (getActions) {
+            parameters.put(FlameGraphDataProvider.TOOLTIP_ACTION_KEY, true);
+        }
+        TmfModelResponse<Map<String, String>> response = provider.fetchTooltip(parameters, new NullProgressMonitor());
+        Map<String, String> tooltip = response.getModel();
+        return (tooltip == null) ? Collections.emptyMap() : tooltip;
+    }
+
+    private static Map<String, Object> getFetchTooltipParameters(long time, long item, @Nullable IOutputElement element) {
+        @NonNull
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(DataProviderParameterUtils.REQUESTED_TIME_KEY, Collections.singletonList(time));
+        parameters.put(DataProviderParameterUtils.REQUESTED_ITEMS_KEY, Collections.singletonList(item));
+        if (element != null) {
+            parameters.put(DataProviderParameterUtils.REQUESTED_ELEMENT_KEY, element);
+        }
+        return parameters;
+    }
+
+    /**
+     * Zoom thread
+     *
+     * @since 1.1
```

Comment:
Remove since comment. Improve comment text to describe what this thread is used for.